### ISSUE TITLE
[Feature Request] - Manage subtask

### ISSUE THREAD

**Issue by codezninja** [dwaring87/rtm-cli#6]
just started getting into RTM. Love that there's a cli tool. Is there any plans on adding support for subtasks?

**Comment by dwaring87** [comment]
I would like to add support for subtasks.  Personally I haven't used them and I'm not sure what the best way to implement their use would be. 

What I'm thinking at the moment is to add an indicator to the main task indicating the number of sub tasks and then adding a new command that will list the sub tasks for a given task.

Do you have any suggestions on how you would like to see this implemented?

**Comment by ghost** [comment]
so thats exactly what I was thinking as well.
```bash
01 (1) Main Task Description [4] <- brackets would describe the number of tasks
```

then we can use a subcommand like 
```bash
sls [task] #list subtasks
sadd [task] #add subtasks
s... etc
```

**Comment by ajmasia** [comment]
Hi @dwaring87, I have been playing with the RTM API this weekend and it seems that the filter `isSubstack:true` does not work.  Returns a `{rev: 'hasCode'}` object. Any news about this? 
I have opened a ticket for support. II think t would be an easy way to get subtasks.

Congratulations on the project, very open and dynamic 😄

**Comment by beauraines** [comment]
I was poking around with the rtm-api and rtm-cli today and it does not look like the RTM API returns a task's subtasks.

I dumped the `rtm-api` `user.get('rtm.tasks.getList')` response to the console and got the following for a task that has 4 sub-tasks. This may be a blocker to this _very useful_ feature as RTM isn't returning that data. 😢 

```json
{
    "id": "6xxxxx6",
    "taskseries": [
        {
            "id": "42xxxxx33",
            "created": "2020-07-23T02:23:24Z",
            "modified": "2022-08-28T07:00:31Z",
            "name": "Take care of the pets",
            "source": "android",
            "url": "",
            "location_id": "",
            "rrule": {
                "every": "1",
                "$t": "FREQ=DAILY;INTERVAL=2;WKST=SU"
            },
            "tags": {
                "tag": [
                    "pets"
                ]
            },
            "participants": [],
            "notes": [],
            "task": [
                {
                    "id": "8xxxxxx86",
                    "due": "2022-08-28T07:00:00Z",
                    "has_due_time": "0",
                    "added": "2022-08-26T07:00:25Z",
                    "completed": "",
                    "deleted": "",
                    "priority": "1",
                    "postponed": "0",
                    "estimate": ""
                }
            ]
        }
    ]
}
```

**Comment by dwaring87** [comment]
Unfortunately, until the RTM API adds subtasks into the API response, there's not much that can be done...

**Comment by scottbilas** [comment]
I've been wanting subtask support as well, and came across this thread: https://www.reddit.com/r/Rememberthemilk/comments/1gxk2qv/is_it_just_me_or_is_their_api_hard_to_use/

It seems the trick is setting `api_version=2`

**Comment by beauraines** [comment]
@scottbilas check out my fork and published to npm `https://www.npmjs.com/package/@beauraines/rtm-cli` 

I have built the indicator for subtasks but not anything relating parents to children tasks.

The way the RTM API returns data is strange. I think if we revised how `rtm-cli` stores the local data we might be able to work with those relationships as opposed to "just another task" but that would be a significant revision to what David had originally built. I'm not ready for that yet.
